src/foam/nanos/boot/Boot.java

```diff
@@ -97,6 +97,17 @@ public Boot(String datadir) {
       factories_.put(sp.getName(), factory);
       logger.info("Registering", sp.getName());
       x.putFactory(serviceName, factory);
+
+      // Register link to the service added to the sub context, allowing the
+      // service to be accessible via both the sub context and the root context.
+      // Eg. "foo/test" nspec can be instantiated by
+      //
+      //      x.cd("foo").get("test");
+      // or
+      //      x.get("foo/test");
+      //
+      if ( x != null )
```

Comment:
x can not be null here, else we'd have a NPE on 99.

Reply:
was a typo, fixed in https://github.com/kgrgreer/foam3/pull/2903/commits/50939e5baba98067bf4a2ab2be335db90ac8695f.